src/site/content/en/blog/2022-state-of-css/index.md

```diff
@@ -0,0 +1,2008 @@
+spread amongst other style blocks. It may seem small in these examples, but it
+can have very large impacts, for both convenience and legibility.
+
+{% Aside 'warning' %}
+Nesting more than four or five levels can become more troublesome 
+than repeating selectors. We advise you to not nest so deep, and instead 
+begin a new selector blog and nest within it.
+{% endAside %}
+
+#### Resources
+
+- Spec:
+  [https://www.w3.org/TR/css-nesting-1/](https://www.w3.org/TR/css-nesting-1/)  
+- PostCSS plugin:
+  [https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-nesting](https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-nesting)  
+- Post by [Bramus](https://twitter.com/bramus): [The future of CSS: Nesting
+  Selectors](https://www.bram.us/2019/03/17/the-future-of-css-nesting-selectors/)
+
+### Scoping styles is really hard
+
+{% BrowserCompat 'css.at-rules.scope' %}
+
+Before `@scope`, many strategies existed because styles in CSS cascade, inherit,
+and are globally scoped by default. These features of CSS are very convenient
+for many things, but for complex sites and applications, with potentially many
+different styles of components, the global space and nature of the cascade can
+make styles feel like they're leaking.
+
+After `@scope`, not only can styles be scoped to only within a context, like a
+class, they can also articulate where the styles end and do not continue to
+cascade or inherit. 
+
+In the following example, [BEM](http://getbem.com/naming/) naming convention
+scoping can be reversed into the actual intent. The BEM selector is attempting
+to scope the color of a `header` element to a `.card` container with naming
+conventions. This requires that the header has this classname on it, completing
+the goal. With `@scope`, no naming conventions are required in order to complete
+the same goal without marking up the header element:
+
+```css
+.card__header {
+  color: var(--text);
+}
+
+/* with @scope becomes */
+
+@scope (.card) {
+  header {
+    color: var(--text);
+  }
+}
+```
+
+Here's another example, less component-specific and more about the global scope
+nature of CSS. Dark and light themes have to coexist inside a stylesheet, where
+order matters in determining a winning style. Usually this means dark theme
+styles come after the light theme; this establishes light as the default and
+dark as the optional style. Avoid the ordering and scope battling with `@scope`:
+
+```css
+​​@scope (.light-theme) {
+  a { color: purple; }
+}
+
+@scope (.dark-theme) {
+  a { color: plum; }
+}
+```
+
+To complete the story here, `@scope` also allows the establishing of where the
+style scope ends. This can't be done with any naming convention or preprocessor;
+it's special and only something CSS built-in to the browser can do. In the
+following example, `img` and `.content` styles are exclusively applied when a
+child of a `.media-block`isand a sibling or parent of `.content`:
```

Comment:
```suggestion
To complete the story here, `@scope` also allows the establishing of where the
style scope ends. This can't be done with any naming convention or preprocessor;
it's special and only something CSS built-in to the browser can do. In the
following example, `img` and `.content` styles are exclusively applied when a
child of a `.media-block` is a sibling or parent of `.content`:
```